driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2177,15 +2208,34 @@ public void runMayThrow() throws InterruptedException, ExecutionException {
                                 submitSchemaRefresh(scc.keyspace, scc.table);
                             break;
                         case DROPPED:
-                            if (scc.table.isEmpty())
-                                manager.metadata.removeKeyspace(scc.keyspace);
-                            else {
+                            if (scc.table.isEmpty()) {
+                                final KeyspaceMetadata removed = manager.metadata.removeKeyspace(scc.keyspace);
+                                if(removed != null) {
+                                    executor.submit(new Runnable() {
```

Comment:
Broadcasting of schema change events si done in the manager's executor so that it doesn't block the I/O thread that received the event notification.